### ISSUE TITLE
--print-traffic not working

### ISSUE THREAD

**Issue by ghost** [rg3/youtube-dl#9365]
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2016.05.01**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

it's just downloading like usual.

**Comment by dstftw** [comment]
Carefully read [new issue template](https://github.com/rg3/youtube-dl/issues/new) and provide all requested information.

**Comment by yan12125** [comment]
It's currently broken on 3.x: http://bugs.python.org/issue26892

**Comment by remitamine** [comment]
it doesn't work only for python 3(tested on python 3.5.1).
on python2:
```
python2 __main__.py -v --print-traffic -F http://www.pbs.org/video/1701025927/
[debug] System config: []
[debug] User config: [u'--external-downloader', u'aria2c', u'--external-downloader-args', u'-x 16', u'-f', u'best[height<=720]/bestvideo[height<=720]+bestaudio', u'--sub-format', u'srt/vtt/best', u'--convert-subs', u'srt', u'--sub-lang', u'en,enUS,en-US,ar,arME', u'--write-sub', u'--no-check-certificate']
[debug] Command-line args: [u'-v', u'--print-traffic', u'-F', u'http://www.pbs.org/video/1701025927/']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.05.01
[debug] Git HEAD: 174aba3
[debug] Python version 2.7.11 - Linux-4.5.1-1-ARCH-x86_64-with-glibc2.2.5
[debug] exe versions: ffmpeg 3.0.2, ffprobe 3.0.2, rtmpdump 2.4
[debug] Proxy map: {}
[pbs] 1701025927: Downloading video info JSON
send: u'GET /videoInfo/1701025927?format=json&type=partner HTTP/1.1\r\nHost: player.pbs.org\r\nAccept-Language: en-us,en;q=0.5\r\nAccept-Encoding: gzip, deflate\r\nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nUser-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20150101 Firefox/44.0 (Chrome)\r\nAccept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7\r\nConnection: close\r\n\r\n'
reply: 'HTTP/1.1 404 NOT FOUND\r\n'
header: Server: nginx
header: Date: Sun, 01 May 2016 18:13:45 GMT
header: Content-Type: text/html
header: Content-Length: 94
header: Connection: close
header: Access-Control-Allow-Origin: *
header: X-PBS-appsvrip: 10.140.176.33
header: X-PBS-appsvrname: pbsplayer-prod-app1
...
```
on python3:
```
python __main__.py -v --print-traffic -F http://www.pbs.org/video/1701025927/
[debug] System config: []
[debug] User config: ['--external-downloader', 'aria2c', '--external-downloader-args', '-x 16', '-f', 'best[height<=720]/bestvideo[height<=720]+bestaudio', '--sub-format', 'srt/vtt/best', '--convert-subs', 'srt', '--sub-lang', 'en,enUS,en-US,ar,arME', '--write-sub', '--no-check-certificate']
[debug] Command-line args: ['-v', '--print-traffic', '-F', 'http://www.pbs.org/video/1701025927/']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.05.01
[debug] Git HEAD: 174aba3
[debug] Python version 3.5.1 - Linux-4.5.1-1-ARCH-x86_64-with-arch
[debug] exe versions: ffmpeg 3.0.2, ffprobe 3.0.2, rtmpdump 2.4
[debug] Proxy map: {}
[pbs] 1701025927: Downloading video info JSON
[pbs] 1701025927: Downloading widget/partnerplayer page
[pbs] 1701025927: Downloading portalplayer page
[pbs] 1701025927: Downloading hls-400-2500k-16x9 video url info
```

**Comment by yan12125** [comment]
My patch is landed in CPython HG (https://hg.python.org/cpython/rev/e0ec68d46464). This ticket can be closed when Python 3.6.0 or Python 3.5.2 is released.

**Comment by yan12125** [comment]
Python 3.5.2 is released, [1] closing.

[1] https://docs.python.org/3.5/whatsnew/changelog.html#python-3-5-2
